src/main/java/com/blackduck/integration/detect/ApplicationUpdater.java

```diff
@@ -289,13 +291,41 @@ private boolean runMainClass(Path jarPath)
                 }
             }
             final Class<?> jarFileArchiveClass = classMap.get("org.springframework.boot.loader.archive.JarFileArchive");
+            if (jarFileArchiveClass == null) {
+                logger.warn("{} Server response or download may be corrupted. Unable to find org.springframework.boot.loader.archive.JarFileArchive class.", LOG_PREFIX);
+                return false;
+            }
             final Constructor<?> jarFileArchiveConstructor = jarFileArchiveClass.getConstructor(File.class);
+            if (jarFileArchiveConstructor == null) {
+                logger.warn("{} Server response or download may be corrupted. Unable to find the constructor of org.springframework.boot.loader.archive.JarFileArchive class.", LOG_PREFIX);
```

Comment:
@gopannabd  The `.getConstructor() `method (and similar calls below) throws NoSuchMethodException ... would getting a null here be possible in cases where the constructor is not found as it says in the log?

Reply:
I am not _that_ confident. Your call!

Reply:
I'm happy to keep the null check since this isn't reproducible at the moment (and writing a test case involves a lot of refactoring) .. just wondering if I should be catching a NoSuchMethodException?

Reply:
Yes, that is fine until we run into it (if ever).